### PR TITLE
[may18] Documentation updates: roadmap, contributing, coding guidelines

### LINKED ISSUE #2014: [may18] Update contributing.md
Contributing.md needs a refresh to detail how to find the correct branch to target your PR and to point folks toward the project pages.

### PR DESCRIPTION
**Pull request by davidkline-ms**
Update roadmap to reflect how master is currently being managed as well as to add release cadence and branch management updates re: vNext

Contributing modified to describe how to find the correct dev branch for master.

Adapt coding guidelines from vNext.

Closes #2014, #2015, #2038

### CODE REVIEW

**Comment by StephenHodgson** [comment]
@davidkline-ms does stringing the issues together like this close them all out automatically?

```
Closes #2014, #2015, #2038 
```

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,6 +22,21 @@ This will let us have a brief discussion about the problem and, hopefully, ident
 
 Note:  If you wish to work on something that already exists on our backlog, you can use that work item as your proposal.  
 
+# Identifing the development branch
+
+The master branch does not accept direct pull requests. Each change must be submitted through an appropriate developement branch. Each upcoming MRTK release has its own development branch (ex: may18_dev).
```
Comment (StephenHodgson) [inline CONTRIBUTING.md]: Is there a setting to prevent people from opening a pull request on a specific branch?

Comment (davidkline-ms) [reply CONTRIBUTING.md]: not sure. I can look

File: CodingGuidelines.md [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+
+### Do:
+
+ ```
+ [flags]
+public enum MyEnum
+{
+    None = 1  << 0,
+    Left = 1  << 1,
+    Right = 1 << 2
```
Comment (StephenHodgson) [inline CodingGuidelines.md]: Let's also add:
```
    Both = Left | Right
```

Comment (davidkline-ms) [reply CodingGuidelines.md]: sure

**Comment by davidkline-ms** [comment]
@StephenHodgson, re: chaining the issue numbers together... I don't know, but I manage the resolved issues manually each release (marking them fixed in stabilization, then closing when we release)

File: CodingGuidelines.md [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+     get{ return myValue2; }
+     set{ myValue2 = value }
+ }
+```
+
+ ### Do:
+
+ ```
+ // Enable field to be configurable in the editor and available externally to other scripts (field is correctly serialized in Unity)
+ [SerializeField]
```
Comment (StephenHodgson) [inline CodingGuidelines.md]: Let's also add an example of a tooltip

File: CodingGuidelines.md [hunk 4]
```diff
@@ -0,0 +1,507 @@
...
+ }
+```
+
+ ### Do:
+
+ ```
+ // Enable field to be configurable in the editor and available externally to other scripts (field is correctly serialized in Unity)
+ [SerializeField] 
+ private float myValue; // <- Notice we co-located the backing field above our corrisponding property.
+ public float MyValue
```
Comment (StephenHodgson) [inline CodingGuidelines.md]: missing summary tags in example. Ideally the public properties summary would match the tooltip of the serialized field.

File: CodingGuidelines.md (hunk 3 above)
Comment (davidkline-ms) [reply CodingGuidelines.md]: Some of this feedback is funny... This is a copy of the coding guidelines from vNext with very few changes (small fixes and a namespace change) :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (davidkline-ms) [reply CONTRIBUTING.md]: doesn't look like we can set that :\

Comment (davidkline-ms) [reply CONTRIBUTING.md]: yes. THAT i can do

**Pull request merged**